care/facility/api/viewsets/asset.py

```diff
@@ -252,6 +263,27 @@ def get_queryset(self):
             queryset = queryset.filter(
                 current_location__facility__id__in=allowed_facilities
             )
+
+        return queryset
+
+    def filter_queryset(self, queryset: QuerySet) -> QuerySet:
```

Comment:
you could do this in the get_queryset, that way you will be able to use this to filter assets